lib/ledger_sync/resource.rb

```diff
@@ -64,17 +74,27 @@ def self.klass_from_resource_type(obj)
       LedgerSync.const_get(LedgerSync::Util::StringHelpers.camelcase(obj))
     end
 
-    def self.reference(name, type)
+    def self.has_one(name, type)
       attribute(name)
-      references[name.to_sym] = type
+      references_one[name.to_sym] = type
+    end
+
+    def self.has_many(name, type)
+      attribute(name)
+      references_many[name.to_sym] = type
+    end
+
+    def self.references_one
+      @references_one ||= {}
     end
 
-    def self.references
-      @references ||= {}
+    def self.references_many
+      @references_many ||= {}
     end
 
     def self.reference_klass(name)
-      references[name.to_sym]
+      references_one[name.to_sym] ||
```

Comment:
This will fail if you have a one and many relationship with the same class.  Not sure if that happens in practice, but in theory could be possible.  May not need to be resolved now, but I'd add a TODO comment here just so we are aware of the potential bug if it arises.  Or actually, better yet, I'd just do something like:

`raise "Class #{name} referenced in both a one and many relationship for the #{self.class.name} resource." if references_one.key?(name.to_sym) && references_many.key?(name.to_sym)`

Perhaps even more correct is to throw an error when building the one and many references on class load so it wouldn't even start.  That's probably what I'd do, otherwise, we can expect weird errors.

Reply:
We talked bit about this earlier. We cant have multiple types of relationship tied to one attribute.